src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/AbstractDiagnosticsCollector.java

```diff
@@ -126,24 +126,9 @@ public void collectDiagnostics(PsiJavaFile unit, List<Diagnostic> diagnostics) {
      * @return true if the given annotation matches the given annotation name and
      *         false otherwise.
      */
-    protected static boolean isMatchedAnnotation(PsiClass unit, PsiAnnotation annotation, String annotationFQName) {
+    protected static boolean isMatchedAnnotation(PsiAnnotation annotation, String annotationFQName) {
         String elementName = annotation.getQualifiedName();
-        if (nameEndsWith(annotationFQName, elementName) && unit != null) {
-            // For performance reason, we check if the import of annotation name is
-            // declared
-            if (isImportedJavaElement(unit, annotationFQName))
```

Comment:
If no code is using `nameEndsWith()` or `isImportedJavaElement()` anymore I believe we should remove the methods as part of this clean-up. It's dead code.

Reply:
Sure @mrglavas. The `isImportedJavaElement()` function is being removed as it is not used, while `nameEndsWith()` is retained since it is used by other functions.